cmd/metal-api/internal/service/machine-service.go

```diff
@@ -2020,6 +2029,8 @@ func (r *machineResource) machineCmd(cmd metal.MachineCommand, request *restful.
 			Description: description,
 		}
 		needsUpdate = true
+	case metal.MachineOnCmd:
+		newMachine.State = metal.MachineState{Sleeping: false}
```

Comment:
The actual power state of a machine is already reported by the metal-bmc to the `/machine/ipmi` endpoint. We can there toggle this flag as well.